Sources/FoundationNetworking/URLSession/URLSession.swift

```diff
@@ -666,6 +682,231 @@ internal extension URLSession {
     }
 }
 
+fileprivate struct Lock<State>: @unchecked Sendable {
```

Comment:
Doesn't have to be a part of this PR, but maybe we should expose `FoundationEssentials.LockedState` as `@_spi` to avoid inventing yet another lock type

Reply:
I think we'll soon be able to use `Mutex` anyway.